### PR TITLE
Update spec.bs to address additional feedback

### PR DESCRIPTION
**Pull request by pythagoraskitty**
We iterate on the spec based on further feedback in issues: [81](https://github.com/WICG/shared-storage/issues/81), [83](https://github.com/WICG/shared-storage/issues/83), [88](https://github.com/WICG/shared-storage/issues/88), [89](https://github.com/WICG/shared-storage/issues/89), [90](https://github.com/WICG/shared-storage/issues/90), [91](https://github.com/WICG/shared-storage/issues/91).

***
<a href="https://pr-preview.s3.amazonaws.com/WICG/shared-storage/pull/85.html" title="Last updated on Jul 31, 2023, 10:54 PM UTC (f9cae6d)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/shared-storage/85/8cdd926...f9cae6d.html" title="Last updated on Jul 31, 2023, 10:54 PM UTC (f9cae6d)">Diff</a>

### CODE REVIEW

**Comment by pythagoraskitty** [comment]
@domfarolino @wanderview PTAL when you can. Thank you!

**Comment by pythagoraskitty** [comment]
Preview link: https://wicg.github.io/shared-storage/pr-preview/refs/pull/85/merge/index.html

**Review by wanderview — COMMENTED** [review]
Overall looks reasonable with some comments.  I suggest waiting for Dominic's review, however.

File: spec.bs
Comment (pythagoraskitty) [reply spec.bs]: Changed to using a specifier in the actual reference (because the compiler complained otherwise).

File: spec.bs
Comment (pythagoraskitty) [reply spec.bs]: Changed to using a specifier in the actual reference (because the compiler complained otherwise).

File: spec.bs
Comment (pythagoraskitty) [reply spec.bs]: Fixed.

File: spec.bs
Comment (pythagoraskitty) [reply spec.bs]: Linked.

File: spec.bs
Comment (pythagoraskitty) [reply spec.bs]: Fixed.

File: spec.bs
Comment (pythagoraskitty) [reply spec.bs]: Fixed.

File: spec.bs
Comment (pythagoraskitty) [reply spec.bs]: Fixed.

**Comment by pythagoraskitty** [comment]
@alexmturner, can you please take a look also when you get a chance, to see if these changes fix the issues you raised?  Thanks

File: spec.bs
Comment (alexmturner) [inline spec.bs:865]: I'm not sure we need to wait here -- I think the [=call=]s above are blocking. Maybe the "[=Queue a global task=] on the [=DOM manipulation task source=], given |realm|'s [=global object=], to resolve |promise| with undefined." needs to be moved earlier?

(ditto for selectURL)

File: spec.bs
Comment (alexmturner) [inline spec.bs:343]: I don't think matches the current implementation. When this is accessed, I think there's instead a DOMException with name NotAllowedError.

**Review by alexmturner — CHANGES_REQUESTED** [review]
Generally seems to fix the issues :) but had a few comments

File: spec.bs
Comment (pythagoraskitty) [reply spec.bs]: done.

File: spec.bs
Comment (pythagoraskitty) [reply spec.bs]: Good point. 

I put a TypeError in the spec for consistency with the rest of the spec (and per a previous comment from wanderview@).

File: spec.bs
Comment (alexmturner) [reply spec.bs:865]: Oh I think you're right -- I don't see an equivalent issue there :)

File: spec.bs
Comment (alexmturner) [reply spec.bs]: Sg! Defer to you and wanderview@, but I'm also wondering two more things: should this return undefined instead of null? And is `sharedStorage` still default initialized if its type has the `?` optional marker?

**Review by alexmturner — APPROVED** [review]

File: spec.bs
Comment (alexmturner) [reply spec.bs]: Ah actually -- just realized that throwing terminates execution (https://infra.spec.whatwg.org/#algorithm-control-flow), so I don't think you need to return null (and so can probably remove the `?` optional marker, I think)

Comment (pythagoraskitty) [reply spec.bs]: Done, thanks.

File: spec.bs
Comment (domfarolino) [inline spec.bs:29]: This term is already exported so I think you should add:

```
spec:html; type:dfn; for:realm; text:global object
```

in the `<pre class=link-defaults>` above

File: spec.bs
Comment (domfarolino) [inline spec.bs:233]: nit: doesn't have to be taken care of in this PR but I'd probably make this `<dfn for=SharedStorageWorklet>addModule initiated</dfn>`

Comment (domfarolino) [inline spec.bs:233]: Later in the spec you link to `[=boolean=]` for associated booleans, but not here. I'd probably try and be consistent (bias towards always linking).

File: spec.bs
Comment (domfarolino) [inline spec.bs:343]: nit: link to https://webidl.spec.whatwg.org/#dfn-throw

File: spec.bs
Comment (domfarolino) [inline spec.bs:575]: Can this instead be `[=this=]'s [=relevant global object=]'s [=Window/browsing context=]` (like in https://wicg.github.io/fenced-frame/#dom-fence-reportevent)? Just reaching for `{{WindowSharedStorage}}` object I think it less good than using `[=this=]`.

this comment applies to the instances just like this below too (there seem to be at least 3 maybe)

File: spec.bs
Comment (domfarolino) [inline spec.bs:793]: nit (outside of this PR's specific changes): `[=fence.reportEvent()=]` should be `{{Fence/reportEvent()}}`

Comment (domfarolino) [inline spec.bs:793]: I think `[=fenced frame=]` should be `<{fencedframe}>`

Comment (domfarolino) [inline spec.bs:793]: nit: I'd personally revert the "(i.e. primary main frame)" change here, since this is specific to Chromium internals terminology and non-normative nor necessarily shared across browsers. You could maybe add this as a note if you'd like (mentioning the Chromium-specific "primary" term) but I might just remove it altogether.

File: spec.bs
Comment (domfarolino) [inline spec.bs:892]: I think instead of "casting" we need to go through the ECMAScript => WebIDL conversion algorithm: https://webidl.spec.whatwg.org/#dfn-convert-ecmascript-to-idl-value

File: spec.bs
Comment (domfarolino) [inline spec.bs:893]: nit: link to https://webidl.spec.whatwg.org/#an-exception-was-thrown

File: spec.bs
Comment (domfarolino) [inline spec.bs:958]: Is there an extra "." before `{{Window}}` here? Is that intentional?

Comment (domfarolino) [inline spec.bs:958]: Can you clarify what "is enabled" means? Does this getter only exist if a permission policy is enabled (and we can then link to the "allowed" algorithm dfn?) Or does this mean something else..

**Comment by pythagoraskitty** [comment]
Realized that the promises weren't being properly rejected when Shared Storage is disabled, so I've fixed that it my latest commit.

Still have several comments to address and some related issues that have since been opened. These are in my queue.

[Preview here](https://wicg.github.io/shared-storage/pr-preview/refs/pull/85/merge/index.html) (Reposting from above)

**Comment by pythagoraskitty** [comment]
I still have some outstanding comments to address. I will open a new PR shortly in order to do so, but I need to land this now for logistical reasons.

**Comment by domfarolino** [comment]
Can you re-open any issues that may have been closed prematurely by the merge, given your last comment?

**Comment by pythagoraskitty** [comment]
Yes, definitely will do!  Although it might be a day or so before I get to it.  Will try to do it this week.

**Pull request merged**
